wire-library/wire-tests-swift/src/main/swift/Person.swift

```diff
@@ -210,13 +238,50 @@ extension Person : Proto2Codable {
 
 #if !WIRE_REMOVE_CODABLE
 extension Person : Codable {
-    public enum CodingKeys : String, CodingKey {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: Person.CodingKeys.self)
+        self.name = try container.decode(String.self, forKey: "name")
+        self.id = try container.decode(Int32.self, forKey: "id")
+        self.email = try container.decodeIfPresent(String.self, forKey: "email")
+        self.phone = try container.decodeIfPresent([Person.PhoneNumber].self, forKey: "phone") ?? []
+        self.aliases = try container.decodeIfPresent([String].self, forKey: "aliases") ?? []
+    }
 
-        case name
-        case id
-        case email
-        case phone
-        case aliases
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: Person.CodingKeys.self)
+
+        try container.encode(self.name, forKey: "name")
+        try container.encode(self.id, forKey: "id")
+        if encoder.protoDefaultValuesEncodingStrategy == .emit || self.email != nil {
```

Comment:
it's actually kinda weird...

> When generating JSON-encoded output from a protocol buffer, if a protobuf field has the default value and if the field doesn't support field presence, it will be omitted from the output by default. An implementation may provide options to include fields with default values in the output.
> 
> A proto3 field that is defined with the optional keyword supports field presence. Fields that have a value set and that support field presence always include the field value in the JSON-encoded output, even if it is the default value.

I think checking `nil` is actually correct? Weirdly that means the `required` fields would should be doing additional checking?

But yeah, I can ask where to look in the channel